src/main.cpp

```diff
@@ -471,11 +471,13 @@ int main(int argc, char** argv) {
     }
 
     if (bForeground) {
-        int iPid = getpid();
-        CUtils::PrintMessage("Staying open for debugging [pid: " +
-                             CString(iPid) + "]");
+        if(CDebug::Debug()) {
+            int iPid = getpid();
+            CUtils::PrintMessage("Staying open for debugging [pid: " +
+                                CString(iPid) + "]");
 
-        pZNC->WritePidFile(iPid);
+            pZNC->WritePidFile(iPid);
```

Comment:
I think this should always write the pid file.

Why don't you just change the message instead? Something like `Running in foreground` or so.